README.rst

```diff
@@ -0,0 +1,591 @@
+                'gretchen',
+                main=partial(say_hello, 'donny'),
+                rpc_module_paths=[_this_module],
+            )
+            print(await gretchen.result())
+            print(await donny.result())
+            await donny.cancel_actor()
+            print("CUTTTT CUUTT CUT!!?! Donny!! You're supposed to say...")
+
+
+    tractor.run(main)
+
+
+Here, we've spawned two actors, *donny* and *gretchen* in separate
+processes. Each starts up and begins executing their *main task*
+defined by an async function, ``say_hello()``.  The function instructs
+each actor to find their partner and say hello by calling their
+partner's ``hi()`` function using a something called a *portal*. Each
+actor receives a response and relays that back to the parent actor (in
+this case our "director").
+
+To gain more insight as to how ``tractor`` accomplishes all this please
+read on!
+
+
+Actor spawning and causality
+----------------------------
+``tractor`` tries to take ``trio``'s concept of causal task lifetimes
+to multi-process land. Accordingly ``tractor``'s actor nursery behaves
+similar to the nursery_ in ``trio``. That is, an ``ActorNursery``
+created with ``tractor.open_nursery()`` waits on spawned sub-actors to
+complete (or error) in the same causal_ way ``trio`` waits on spawned
+subtasks. This includes errors from any one sub-actor causing all other
+actors spawned by the nursery to be cancelled_.
+
+To spawn an actor open a *nursery block* and use the ``start_actor()``
+method:
+
+.. code:: python
+
+    def movie_theatre_question():
+        """A question asked in a dark theatre, in a tangent
+        (errr, I mean different) process.
+        """
+        return 'have you ever seen a portal?'
+
+
+    async def main():
+        """The main ``tractor`` routine.
+        """
+        async with tractor.open_nursery() as n:
+            portal = await n.start_actor(
+                'frank',
+                # enable the actor to run funcs from this current module
+                rpc_module_paths=[__name__],
+                outlive_main=True,
+            )
+
+            print(await portal.run(__name__, 'movie_theatre_question'))
+            # calls the subactor a 2nd time
+            print(await portal.run(__name__, 'movie_theatre_question'))
+
+            # the async with will block here indefinitely waiting
+            # for our actor "frank" to complete, but since it's an
+            # "outlive_main" actor it will never end until cancelled
+            await portal.cancel_actor()
+
+Notice the ``portal`` instance returned from ``nursery.start_actor()``,
+we'll get to that shortly.
+
+Spawned actor lifetimes can be configured in one of two ways:
+
+- the actor terminates when its *main* task completes (the default if
+  the ``main`` kwarg is provided)
+- the actor can be told to ``outlive_main=True`` and thus act like an RPC
+  daemon where it runs indefinitely until cancelled
```

Comment:
Okay, this is something to talk about in person, but from this reading, I see dubious value to `outlive_main`.

Reply:
Or, you're just assuming the reader will understand why its useful.

You've just brought it up casually while talking about benefit cancel scopes when it seems to subvert it. Its also then even weirder that you lead with that particular example as well.

Reply:
See #25 and #24!